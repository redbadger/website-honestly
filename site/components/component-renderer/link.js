// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
