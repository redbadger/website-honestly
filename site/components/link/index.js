// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
