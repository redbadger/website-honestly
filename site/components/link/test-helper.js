// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
