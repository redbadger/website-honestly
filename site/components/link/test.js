// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
