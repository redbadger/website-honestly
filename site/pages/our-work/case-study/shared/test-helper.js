// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
