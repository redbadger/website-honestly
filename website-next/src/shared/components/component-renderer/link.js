import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
