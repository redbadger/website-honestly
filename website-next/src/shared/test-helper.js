import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
