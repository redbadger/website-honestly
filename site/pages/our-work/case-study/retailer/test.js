import React from 'react';
import { render } from 'enzyme';

import { Context } from '../../../../components/link/test-helper';
import CaseStudyRetailer from './index';

jest.mock(
  '../shared/what-to-read-next',
  () => require.requireActual('../shared/test-helper').default,
);

describe('site/pages/our-work/case-study/retailer/', () => {
  it('renders correctly', () => {
    expect(
      render(
        <Context>
          <CaseStudyRetailer />
        </Context>,
      ),
    ).toMatchSnapshot();
  });
});
