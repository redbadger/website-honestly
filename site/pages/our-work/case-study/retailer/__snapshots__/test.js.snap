// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/our-work/case-study/retailer/ renders correctly 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="header__container"
    >
      <div
        class="header__imageContainer"
      >
        <img
          alt="Toy shopping cart filled with fruit & veg, surrounded by key statistics "
          class="header__image"
          src="header.jpg"
        />
      </div>
      <div
        class="header__outcome header__outcome--right"
      >
        <span
          class="header__outcome__percentage"
        >
          +54%
        </span>
        <h3
          class="header__outcome__description"
        >
          Increased orders
        </h3>
      </div>
      <div
        class="header__outcome header__outcome--left"
      >
        <span
          class="header__outcome__percentage"
        >
          +29%
        </span>
        <h3
          class="header__outcome__description"
        >
          Uplift in visits
        </h3>
      </div>
    </div>
  </div>,
  <div
    class="body"
  >
    <div
      class="heading"
    >
      <h1
        class="fontL "
      >
        Next generation platform for retail giant cleared five-year backlog in just eight months
      </h1>
    </div>
    <div
      class="section"
    >
      <h2
        class="sectionHeading"
      >
        <span
          class="redTitle"
        >
          Let's make things better
        </span>
        Solving high drop-out rates
      </h2>
      <div
        class="sectionBody"
      >
        <p
          class="fontXS paragraph"
        >
          Our client, one of the largest retailers in the UK was experiencing lost revenue in the millions, due to higher than expected drop-out rates on its online platform.
        </p>
        <p
          class="fontXS paragraph"
        >
          This was because the online experience and functionality were inconsistent. The business was spending huge sums on a piece of software that scraped its website, and then created mobile optimised versions on the fly. This meant that some orders were easier to place than others - depending on what device was being used - and amendments to orders were down as they were incredibly complicated to change once booked.
        </p>
        <p
          class="fontXS paragraph"
        >
          As a retailer dedicated to providing value to customers and seeing its online business growing substantially - including a 100% year-on-year mobile traffic increase - there was an urgent need for the site to work seamlessly. The business needed to build a modern technology platform using cutting-edge, web-based tech to ensure all experiences for customers were intuitive and easy to use, regardless of device. This would also have to factor in processing a catalogue of hundreds-of-thousands of products, delving into complex user journeys, and building a well executed international shipping function.
        </p>
        <p
          class="fontXS paragraph"
        >
          As a business that championed innovation this retailer decided that Red Badger was the right partner to help it make things better.
        </p>
      </div>
    </div>
    <div
      class="section"
    >
      <h2
        class="sectionHeading"
      >
        <span
          class="redTitle"
        >
          Do the right thing. Do the thing right.
        </span>
        Streamlining teams and implementing cutting-edge tech
      </h2>
      <div
        class="sectionBody"
      >
        <h3
          class="fontS2 "
        >
          Embracing organisational change
        </h3>
        <p
          class="fontXS paragraph"
        >
          Although change is happening, big companies are still watching the speed at which today’s start-ups move in awe and envy. What start-ups do right is work in a way that puts the customer at the very heart of projects, from technology choices to design and content. They then work in a collaborative and Lean way to get new products out at speeds that seem staggering to large organisations.
        </p>
        <p
          class="fontXS paragraph"
        >
          Red Badger enabled this large retail client to work with the agility of a start-up: re-engineering and revolutionising its workflow with Lean practices and helping it break down department silos by keeping design, development and testing in one place. The benefits of this included:
        </p>
        <ul
          class="list"
        >
          <li
            class="content__list__element"
          >
            A shorter feedback loop
          </li>
          <li
            class="content__list__element"
          >
            Team-wide responsibility for each decision
          </li>
          <li
            class="content__list__element"
          >
            Elimination of waste
          </li>
          <li
            class="content__list__element"
          >
            Driven determination and focus in order to deliver value to customers, faster
          </li>
          <li
            class="content__list__element"
          >
            A sped-up delivery, over four years early
          </li>
        </ul>
      </div>
    </div>
    <div
      class="section"
    >
      <picture>
        <source
          class="techGraphImage"
          srcset="tech-graph.jpg"
        />
        <img
          alt="Tech graph"
          class="techGraphImage"
          src="tech-graph.jpg"
        />
      </picture>
    </div>
    <div
      class="section"
    >
      <div
        class="sectionBody"
      >
        <h3
          class="fontS2 "
        >
          Finding the right tech for the job
        </h3>
        <p
          class="fontXS paragraph"
        >
          Now working in a collaborative and Lean way with Red Badger, it was time to find the right tech for the job.
        </p>
        <p
          class="fontXS paragraph"
        >
          Red Badger worked very closely with the retailer’s engineering team to recommend and introduce an incredibly progressive selection of cutting-edge technology. The new mobile site had to replace the existing screen-scraping solution and improve the customer experience, making the new site adaptable, flexible and able to scale up to the ever-growing demand that the business was seeing.
        </p>
      </div>
    </div>
    <div
      class="contactBox"
    >
      <h2
        class="contactBox__heading"
      >
        Project in mind?
      </h2>
      <button
        class="contactBox__button"
        type="button"
      >
        Tell us more
      </button>
    </div>
    <div
      class="section"
    >
      <div
        class="sectionBody"
      >
        <p
          class="fontXS paragraph"
        >
          By using a combination of Node.js and React.js, the team were able to develop an app that delivers a consistent experience across all devices. It also means that the application is fast and responsive, delivering far better usability for customers.
        </p>
        <p
          class="fontXS paragraph"
        >
          On top of this, by moving into the Cloud, using advanced technology such as Docker Containers, AWS ECS and Terraform, the ecommerce store was able to incorporate auto cloud scaling which expands and contracts resource allocation to match customer demand. This is vital in helping to cope with the huge growth in revenue and transaction volumes experienced by the retailer, especially around times of peak demand such as Christmas and Black Friday.
        </p>
        <p
          class="fontXS paragraph"
        >
          Finally, on the tech front, by building Automated Test and Continuous Deployment Pipelines, new features are enabled to be constantly trickled into production, continually delivering new value to customers and substantially reducing risk. Previously, the retail giant released quarterly but can now release multiple times a day with deployments taking minutes, rather than days.
        </p>
      </div>
    </div>
    <div
      class="section"
    >
      <h2
        class="sectionHeading"
      >
        <span
          class="redTitle"
        >
          Creating lasting change.
        </span>
        Improving conversion rates by 83%
      </h2>
      <div
        class="sectionBody"
      >
        <p
          class="fontXS paragraph"
        >
          Successfully using the technologies above has solved long-standing issues and fundamentally influenced the retailer’s next generation platform. Where legacy systems previously ruled, cutting-edge innovation allows its online business to grow unencumbered by technical limitations. The new platform now provides this retailer with the means to scale for years to come.
        </p>
        <p
          class="fontXS paragraph"
        >
          And, by first focusing on delivering the application for mobile only, Red Badger has delivered a completed application across seven countries in just eight months, the results of which are staggering:
        </p>
      </div>
    </div>
    <div
      class="section"
    >
      <div
        class="Card overview"
      >
        <div
          class="listBox listBox"
        >
          <h3
            class="listBox__heading"
          >
            What we did
          </h3>
          <ul>
            <li
              class="listBox__element listBox__element--results"
            >
              <div
                class="listBox__leftLabel"
              >
                Mobile first application across 7 countries
              </div>
              <div />
            </li>
            <li
              class="listBox__element listBox__element--results"
            >
              <div
                class="listBox__leftLabel"
              >
                Introduced Lean & Agile practices
              </div>
              <div />
            </li>
            <li
              class="listBox__element listBox__element--results"
            >
              <div
                class="listBox__leftLabel"
              >
                Built Continuous Deployment Pipelines
              </div>
              <div />
            </li>
            <li
              class="listBox__element listBox__element--results"
            >
              <div
                class="listBox__leftLabel"
              >
                Moved to the Cloud
              </div>
              <div />
            </li>
            <li
              class="listBox__element listBox__element--results"
            >
              <div
                class="listBox__leftLabel"
              >
                Introduced automated testing & cutting edge tech
              </div>
              <div />
            </li>
          </ul>
        </div>
        <div
          class="listBox listBox"
        >
          <h3
            class="listBox__heading"
          >
            Results
          </h3>
          <ul>
            <li
              class="listBox__element listBox__element--results"
            >
              <div>
                Uplift in visits
              </div>
              <div>
                +29%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results"
            >
              <div>
                Increased orders
              </div>
              <div>
                +54%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results"
            >
              <div>
                Customers amending orders
              </div>
              <div>
                +443%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results"
            >
              <div>
                Conversion rates including amendments
              </div>
              <div>
                +83%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results"
            >
              <div>
                Conversion rates excluding amendments
              </div>
              <div>
                +18%
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  <div
    class="whatNext"
  >
    <div
      class="whatNext__tilesContainer"
    >
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/pride/"
          >
            <img
              alt="Pride in London"
              class="whatNext__image"
              src="pride.gif"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Pride in London
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/pride/"
        >
          <h2
            class="whatNext__title"
          >
            Delivering a best-in-class digital experience
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/financial-times/"
          >
            <img
              alt="Financial Times"
              class="whatNext__image"
              src="financialTimesReadMore.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Financial Times
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/financial-times/"
        >
          <h2
            class="whatNext__title"
          >
            Lasting change for a media giant
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/car-trawler/"
          >
            <img
              alt="Car Trawler"
              class="whatNext__image"
              src="carTrawler.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Car Trawler
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/car-trawler/"
        >
          <h2
            class="whatNext__title"
          >
            Creating complete CMS control for travel brands
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
    </div>
    <a
      class="whatNext__button"
      href="/our-work/"
    >
      See more of our work
    </a>
  </div>,
  <section
    class="contactUsContainer"
    data-cy="contact-us-slice"
    id="contactUs"
  >
    <h2
      class="header"
    >
      We can help you
    </h2>
    <div
      class="contentContainer"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          Create & validate new ideas
        </li>
        <li
          class="item"
        >
          Deliver great quality products & services, fast
        </li>
        <li
          class="item"
        >
          Be bold with technology
        </li>
        <li
          class="item"
        >
          Be more customer centric
        </li>
        <li
          class="item"
        >
          Improve efficiency with lean practices
        </li>
        <li
          class="item"
        >
          Build capability & confidence
        </li>
      </ul>
      <div
        class="imgContainer"
      />
    </div>
    <div>
      <a
        class="mailToLink"
        data-cy="email-link"
        href="mailto:hello@red-badger.com?Subject=Can%20you%20help%20me%20with%20..."
      >
        Send an email
      </a>
      <span
        class="talkToUs"
      >
        hello@red-badger.com
      </span>
    </div>
  </section>,
]
`;
